ovs/lib/storagerouter.py

```diff
@@ -328,38 +341,71 @@ def add_vpool(parameters):
+                    error_messages.append('Missing the connection information for the Fragment Cache Backend')
                 else:
                     try:
                         Toolbox.verify_required_params(actual_params=parameters,
-                                                       required_params={'cache_quota': (float, {'min': 0.1 * 1024.0 ** 3, 'max': 1 * 1024.0 ** 4}, False),
-                                                                        'connection_info_aa': (dict, {'host': (str, Toolbox.regex_ip),
+                                                       required_params={'cache_quota_fc': (float, {'min': 0.1 * 1024.0 ** 3, 'max': 1 * 1024.0 ** 4}, False),
+                                                                        'connection_info_fc': (dict, {'host': (str, Toolbox.regex_ip),
                                                                                                       'port': (int, {'min': 1, 'max': 65535}),
                                                                                                       'client_id': (str, None),
                                                                                                       'client_secret': (str, None),
                                                                                                       'local': (bool, None, False)})})
                         # Check cache quota is large enough for at least 10 vDisks
-                        ovs_client = OVSClient(ip=connection_info_aa['host'],
-                                               port=connection_info_aa['port'],
-                                               credentials=(connection_info_aa['client_id'], connection_info_aa['client_secret']),
-                                               version=2)
-                        cache_quota = parameters.get('cache_quota')
+                        cache_quota = parameters.get('cache_quota_fc')
                         if cache_quota is not None:
-                            backend_dict_aa = ovs_client.get('/alba/backends/{0}/'.format(alba_backend_guid_aa), params={'contents': 'name,usages'})
-                            if backend_dict_aa['usages']['free'] < cache_quota * 10:
-                                raise RuntimeError('Requested cache quota is too high, please lower the quota or add more ASDs to ALBA Backend {0}'.format(backend_dict_aa['name']))
+                            ovs_client = OVSClient(ip=connection_info_fc['host'],
+                                                   port=connection_info_fc['port'],
+                                                   credentials=(connection_info_fc['client_id'],
+                                                                connection_info_fc['client_secret']),
+                                                   version=2)
+                            backend_dict_fc = ovs_client.get('/alba/backends/{0}/'.format(alba_backend_guid_fc), params={'contents': 'name,usages'})
```

Comment:
If the `alba_backend_guid_fc` is identical to `alba_backend_guid_bc` this gets executed twice